src/canvas-extractor/edfi_canvas_extractor/client_graphql.py

```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+import sqlalchemy
+
+from pandas import DataFrame
+from typing import List, Tuple
+
+from canvasapi.course import Course
+from canvasapi.section import Section
+
+from edfi_canvas_extractor.graphql.extract import Extract
+from edfi_canvas_extractor.graphql import (
+    courses as coursesGQL,
+    sections as sectionsGQL,
+)
+from edfi_canvas_extractor.mapping import (
+    sections as sectionsMap,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_courses(
+    gql: Extract,
+    sync_db: sqlalchemy.engine.base.Engine,
+) -> Tuple[List[Course], DataFrame]:
+    """
+    Gets all Canvas courses for the given date range, in the Ed-Fi UDM format.
+
+    Parameters
+    ----------
+    canvas: Canvas
+        Canvas object.
+    start_date: str
+        Retrieve Courses starting on or after this date.
+    end_date:
+        Retrieve Courses ending on or before this date.
+    sync_db: sqlalchemy.engine.base.Engine
+        Sync database connection.
```

Comment:
Parameters block does not align with actual function parameters. I assume you'll review all of the document blocks before the final code review; therefore I won't comment on any others.